assert/assertions.go

```diff
@@ -901,25 +901,15 @@ func Condition(t TestingT, comp Comparison, msgAndArgs ...interface{}) bool {
 type PanicTestFunc func()
 
 // didPanic returns true if the function passed to it panics. Otherwise, it returns false.
-func didPanic(f PanicTestFunc) (bool, interface{}) {
+func didPanic(f PanicTestFunc) (didPanic bool, message interface{}) {
+	didPanic = true
 
-	didPanic := false
-	var message interface{}
-	func() {
-
-		defer func() {
-			if message = recover(); message != nil {
-				didPanic = true
-			}
-		}()
-
-		// call the target function
-		f()
-
-	}()
-
-	return didPanic, message
+	defer func() { message = recover() }()
+	// call the target function
+	f()
+	didPanic = false
 
+	return
```

Comment:
Is this line not redundant/pointless now?

Reply:
I think it's required when returning named return values.